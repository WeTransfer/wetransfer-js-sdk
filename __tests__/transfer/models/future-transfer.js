const { futureTransfer } = require('../../../src/transfer/models');

describe('Future transfer normalizer', () => {
  describe('normalizeTransfer function', () => {
    let transfer = {};
    beforeEach(() => {
      transfer = {
        name: 'WeTransfer rocks',
        description: '',
      };
    });

    it('should return a normalized transfer', () => {
      const normalized = futureTransfer(transfer);
      expect(normalized).toMatchSnapshot();
    });

    it('should remove extra properties', () => {
      const extraProps = Object.assign({}, transfer, { date: new Date() });
      const normalized = futureTransfer(extraProps);
      expect(normalized).toMatchSnapshot();
    });

    it('add default values', () => {
      const normalized = futureTransfer({});
      expect(normalized).toMatchSnapshot();
    });
  });
});
