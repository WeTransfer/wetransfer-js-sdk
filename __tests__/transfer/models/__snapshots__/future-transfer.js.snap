// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Future transfer normalizer normalizeTransfer function add default values 1`] = `
Object {
  "description": "",
  "name": "",
}
`;

exports[`Future transfer normalizer normalizeTransfer function should remove extra properties 1`] = `
Object {
  "description": "",
  "name": "WeTransfer rocks",
}
`;

exports[`Future transfer normalizer normalizeTransfer function should return a normalized transfer 1`] = `
Object {
  "description": "",
  "name": "WeTransfer rocks",
}
`;
